### ISSUE TITLE
New Tab Page (#GitCoin Funded)

### ISSUE THREAD

**Issue by AdamSC1-ddg** [ironcoinext/IronCoin#8]
We'll design a new tab page which will:

- Use the Coingecko.com or Messari API to show cards of the top three coin prices. Users can customize which three coins are shown in the cards, and this preference should be stored in localstorage.
- Display the number of sites blocked by IronCoin, the number of punycode attacks in our database and the number of scam sites in our database.
- Have a link to report a site.
- Show a quick table of the top 100 coins prices (similar to Coinmarketcap.com), also from the Coingecko.com API or Messari API.

**To claim this GitCoin Bounty a contributor must:**

1. Design and implement a visually pleasing new tab page, integrated with a third-party API.
2. Build a well documented version of that system that can be merged into the main extension code without failing.
3. Successfully pass a review by two main contributors to the project.
4. Ensure that the load time does not interfere with the user experience. (Maximum 0.5 seconds, target 0.25 seconds)
6. Be fully implemented into the new code with documentation for future maintainers. 

If at any time the contributing developer cannot pass a stage of the review and integration process then a partial bounty or no bounty will be paid out.

**Other important considerations:**

_Compatibility:_

* The implementation methods chosen should be ones that will work with the extension APIs in Chrome (and Chromium based browsers), as well as Firefox and Safari. Extra consideration will be given if the methods work in other browser environments, but, support for Chrome, Firefox and Safari is the minimum requirement.

_Client Side Security:_

* There should be no need for any functionality to exist beyond the client side of the extension in this issue.

_Speed:_

* Page speed for a new tab page should be under 0.5 seconds, and ideally move towards 0.25 seconds as a target. If due to third-party API delays this is impossible, we must show that all possible changes on our end have been made to get as close to 0.5 seconds as possible.

_Conflict:_

* This section is not applicable for this issue.

_Whitelist:_

* This section is not applicable for this issue.

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 1.0 ETH (206.37 USD @ $206.37/ETH)  attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253).
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $131,190.03 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by x5engine** [comment]
I would like to implement this new tab with all the needed tasks here.

I can design this tab with all API data since I have built my own API (EmbedAPI.com) so this can be easily done.

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 5 days, 1 hour ago.
Please review their action plans below:


**1) [aerophile](https://gitcoin.co/profile/aerophile) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?mutate_worker_action=approve&worker=aerophile) | [reject worker](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?mutate_worker_action=reject&worker=aerophile))_.**

Hi there,
I'm interested in taking this up. I'll like the detailed level of requirements that has been provided with this task and would aim to deliver this within days.  Looking forward to starting work on this as soon as I'm approved. cheers.
**2) [lilstronghands](https://gitcoin.co/profile/lilstronghands) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?mutate_worker_action=approve&worker=lilstronghands) | [reject worker](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?mutate_worker_action=reject&worker=lilstronghands))_.**

create an aesthetically appealing page connected to the coingecko api to show 3 coins prices, default the top 3 coins and allow the user to choose if they want to change any of the coins to others listed on coingecko. Display the number of sites blocked by ironCoin along with the amount of punycode attacks in the database and the number of scam sites in said database. add a link to report the site along with a table of the top 100 coins.
**3) [alex-coding-c](https://gitcoin.co/profile/alex-coding-c) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?mutate_worker_action=approve&worker=alex-coding-c) | [reject worker](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?mutate_worker_action=reject&worker=alex-coding-c))_.**

Feel free to check my github profile.
github.com/alex-coding-c
**4) [mushroomgenie](https://gitcoin.co/profile/mushroomgenie) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?mutate_worker_action=approve&worker=mushroomgenie) | [reject worker](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?mutate_worker_action=reject&worker=mushroomgenie))_.**

I would like to work on this issue. I have already setup a developer environment for IronCoin and made PR regarding it. 
https://github.com/ironcoinext/IronCoin/pull/9
I have a few doubts regarding the design of the new tab page, it would be helpful if I could get some more insight on the design.
**5) [vaibhavdaren](https://gitcoin.co/profile/vaibhavdaren) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?mutate_worker_action=approve&worker=vaibhavdaren) | [reject worker](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?mutate_worker_action=reject&worker=vaibhavdaren))_.**

I would like to take up this issue.
**6) [anish-agnihotri](https://gitcoin.co/profile/anish-agnihotri) has been approved to start work.**

Hey there!

My name is Anish and I'm a Gitcoin bounty hunter who has experience building components (tab page in this case) like this. I'm actually currently working on a very similar [Libra Explorer](https://github.com/librasearch), so I know my way around.

I'll hook up the API's either through a direct REST component call and make them query on extension usage, or I'll use a WebSocket so that we have real-time updating for the required data.

ETA on completion should be within the next two days, and I'll keep everyone updated through this Github issue if approved!

Thanks,

Anish
**7) [domantasp](https://gitcoin.co/profile/domantasp) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?mutate_worker_action=approve&worker=domantasp) | [reject worker](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?mutate_worker_action=reject&worker=domantasp))_.**

My work plan:

1. Designing the page using HTML and CSS.
2. Connecting to the appropriate API to show real information.
3. Storing user's preference in the local storage.
4. Add counter for blocked sites.
5. Optimize for performance.
6. Document the code.

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253).

**Comment by Anish-Agnihotri** [comment]
Forgot to mention in my application but feel free to look through my [Github](https://github.com/anish-agnihotri) or [Gitcoin](https://gitcoin.co/profile/anish-agnihotri) profiles for past programming + bounty experience, and my [Dribbble](https://dribbble.com/anishagnihotri) for design experience (since the issue requires a fair bit of UI + UX input).

**Comment by gitcoinbot** [comment]
@anish-agnihotri Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] reminder (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=1) | [3 days](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=3) | [5 days](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=5) | [10 days](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=10) | [100 days](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=100)

**Comment by Anish-Agnihotri** [comment]
This is odd. I never received a notification that I had been approved to begin work from Gitcoin. I’ll bring this up with the team in the Slack as well.

My complete apologies. I’ll finish this this weekend and submit a PR first thing Sunday.

**Comment by Anish-Agnihotri** [comment]
Hi @AdamSC1-ddg,

I'm working on this bounty right now and just wanted to confirm a couple of things:

1. For the "new tab page", did you mean another tab page in the extension, or an overhaul of the existing page?
2. I have the number of punycode attacks (just a loop through the database checking for number of strings containing `xn--`), and the number of scam sites, but how did you want me to get the number of sites blocked by IronCoin? Do you want me to just add to a variable in localstorage everytime a website in the database is accessed and the warning shows up?

Additionally, please take a look at the first draft of the design I quickly made. Of course, things are bound to change so this is just super brief while I test. This example includes two tabs (one for the statistics, and one for the market page with top coins). Let me know your thoughts, etc. 

<img width="442" alt="Screen Shot 2019-08-25 at 5 11 24 pm" src="https://user-images.githubusercontent.com/4398624/63655907-d7eb8a80-c75b-11e9-8926-39a5f0f90747.png">

Note, ofc, this is rough, so I have yet to fix the logo, proper sizings, footer links, colors (from the hard blues to the IronCoin greens), etc.

Let me know your thoughts and I'll go from there.

**Comment by AdamSC1-ddg** [comment]
Hey @Anish-Agnihotri - I think there is a mix up.

We meant a new tab page: https://developer.chrome.com/extensions/override

Not the dropdown from the extension.

Basically this is a page that will appear when the user opens a new tab in Chrome instead of the default landing homepage. The content we want is too much to squish it all in on the extension dropdown.

**Comment by Anish-Agnihotri** [comment]
Oh! I see. That makes a lot more sense. I was initially confused since there was so much content to fit into an extension page.

I'll setup the the default landing page and content tonight and get back to you with a revised version asap.

**Comment by gitcoinbot** [comment]
@anish-agnihotri Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] reminder (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=1) | [3 days](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=3) | [5 days](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=5) | [10 days](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=10) | [100 days](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=100)

**Comment by gitcoinbot** [comment]
@anish-agnihotri Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] reminder (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=1) | [3 days](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=3) | [5 days](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=5) | [10 days](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=10) | [100 days](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=100)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

@anish-agnihotri due to inactivity, we have escalated [this issue](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253) to Gitcoin's moderation team. Let us know if you believe this has been done in error!

* [x] reminder (3 days)
* [x] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=1) | [3 days](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=3) | [5 days](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=5) | [10 days](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=10) | [100 days](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=100)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

@anish-agnihotri due to inactivity, we have escalated [this issue](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253) to Gitcoin's moderation team. Let us know if you believe this has been done in error!

* [x] reminder (3 days)
* [x] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=1) | [3 days](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=3) | [5 days](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=5) | [10 days](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=10) | [100 days](https://gitcoin.co/issue/ironcoinext/IronCoin/8/3253?snooze=100)

**Comment by AdamSC1-ddg** [comment]
Hi @Anish-Agnihotri - is this work still in progress?
